staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -255,10 +277,14 @@ func (a *OIDCAuthenticator) AuthenticateToken(value string) (user.Info, bool, er
 		if !emailVerified {
 			return nil, false, errors.New("email not verified")
 		}
-		username = claim
 	default:
-		// For all other cases, use issuerURL + claim as the user name.
-		username = fmt.Sprintf("%s#%s", a.issuerURL, claim)
+		if a.usernamePrefix == "" {
+			username = fmt.Sprintf("%s#%s", a.issuerURL, username)
+		}
+	}
+
+	if a.usernamePrefix != "" && a.usernamePrefix != "-" {
```

Comment:
probably we should define `-` as a const. or i have to go to the flag help string to figure out why `-` is special.